src/components/Proposal/Create/SchemeForms/CreateGenericMultiCallProposal.tsx

```diff
@@ -135,6 +135,14 @@ class CreateGenericMultiCallScheme extends React.Component<IProps, IStateProps>
     };
   }
 
+  componentDidUpdate(prevProps: Readonly<IProps>) {
```

Comment:
That's a specific solution for `multiCallGenericScheme`.
A generic solution is better, that will re-render the form when we switch.

Reply:
How does it work?

Reply:
What do you mean? 
Maybe I'm wrong but as I see this, the problem is when you switch in the select to a different scheme (from the same type) the form is not re-rendered. If it's the problem so it should be fixed generically maybe here `SelectProposal.tsx`.
Because this is very specific to `multiCallGenericScheme`.